mkosi/__init__.py

```diff
@@ -3302,6 +3302,12 @@ def build_image(context: Context) -> None:
         output_base.unlink(missing_ok=True)
         output_base.symlink_to(context.config.output_with_compression)
 
+    if context.config.output_mode is not None and context.config.output_format != OutputFormat.directory:
+        os.chmod(output_base, context.config.output_mode)
+        if context.config.output_format in (OutputFormat.uki, OutputFormat.esp):
+            os.chmod(context.staging / context.config.output_split_kernel, context.config.output_mode)
+            os.chmod(context.staging / context.config.output_split_initrd, context.config.output_mode)
```

Comment:
We should blanket apply it to every output I think so I'd do this differently. In `finalize_staging()`, before we copy a file to the output directory, make sure it has the requested output mode.

Reply:
ok, PTAL